docs/rfcs/002-branch-management.md

```diff
@@ -0,0 +1,95 @@
+# Branch Management From Command Palette
+
+## Status
+
+Proposed
+
+## Summary
+
+Efficient branch management that is in line with existing Atom conventions.
+
+## Motivation
+
+Current branch management is incomplete (e.g. no deleting of branches) and not consistent with other conventions found in the Atom UI for comparable actions. It is hard to switch or create new branches with just keyboard input.
+
+## Explanation
+
+The new branch selector exists largely in the [command palette](https://github.com/atom/command-palette). The only UI element is the pre-existing `branch-view` tile in the status bar:
+
+![existing branch view tile](https://cldup.com/CE_61W1uBk.png)
+
+### Command Palette
+
+The command palette contains the following new commands:
+- Create new branch (`github:create-new-branch`)
+- Checkout branch (`github:checkout-branch`)
+- Delete branch (`github:delete-branch`)
+- Rename branch (`github:rename-branch`)
+
+#### `github:create-new-branch`
+
+Executing `github:create-new-branch` opens a text input dialog, similar to when creating a new file asks for the filename. When pressing "Escape", the dialog closes without creating a branch, when pressing "Return" a branch of that name is created and switched to.
+
+![Branch name input](https://cldup.com/GhKgFE_zTy.png)
+
+#### `github:checkout-branch`
+
+Executing `github:checkout-branch` opens a filterable command palette list of local branches, similar to the list of available grammars when executing `grammar-selector:show`. Selecting a branch from that list switches to that branch.
+
+![Local branch selector](https://cldup.com/GYfUP3TG8O.png)
+
+
+#### `github:delete-branch`
+
+Executing `github:delete-branch` opens a filterable command palette list of local branches, the same as executing `github:checkout-branch`. Selecting a branch from that list will show a prompt asking for confirmation to delete _[named branch_], including information whether that branch's merge status (e.g. _'test3' has been merged to 'refs/remotes/origin/test3', but not yet merged to HEAD._).
```

Comment:
I like having the extra merge status in the list here. It's a nice way to give you a little more context about the action you're about to take.

I'm not sure what this looks like in a world where the branch commands are unified. Any ideas, @simurai? Maybe we always show the merge status if there is one... ?

Reply:
Have a different icon?

![image](https://user-images.githubusercontent.com/378023/39225339-bd5a891e-4886-11e8-91c8-764e58bbee88.png)

The merged branch icon is used for [merged pull requests](https://github.com/atom/github/pulls?q=is%3Apr+is%3Aclosed), but maybe ok in this context.

Maybe even less contrast to signal: "This branch can probably be deleted".

![image](https://user-images.githubusercontent.com/378023/39225507-bf6ec5a2-4887-11e8-98ca-d070e2a050a8.png)